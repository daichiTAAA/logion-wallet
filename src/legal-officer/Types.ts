import { ColorTheme, rgbaToHex } from '../component/ColorTheme';
import Identity from '../component/types/Identity';
import PostalAddress from '../component/types/PostalAddress';

export const DEFAULT_LEGAL_OFFICER = "5GrwvaEF5zXb26Fz9rcQpDWS57CtERHpNehXCPcNoHGKutQY"; // Alice
export const ANOTHER_LEGAL_OFFICER = "5FHneW46xGXgs5mUiveU4sbTyGBzmstUspZC92UhjJM694ty"; // Bob

const LEGAL_OFFICERS: string[] = [ DEFAULT_LEGAL_OFFICER, ANOTHER_LEGAL_OFFICER ];

export function isLegalOfficer(address: string): boolean {
    return LEGAL_OFFICERS.includes(address);
}

export type TokenizationRequestStatus = "PENDING" | "REJECTED" | "ACCEPTED";

export interface AssetDescription {
    assetId: string,
    decimals: number,
}

export interface TokenizationRequest {
    id: string,
    legalOfficerAddress: string,
    requesterAddress: string,
    requestedTokenName: string,
    bars: number,
    status: TokenizationRequestStatus,
    rejectReason?: string | null,
    createdOn?: string,
    decisionOn?: string,
    assetDescription?: AssetDescription,
}

export type LegalOfficerDecisionStatus = "PENDING" | "REJECTED" | "ACCEPTED";

export interface LegalOfficerDecision {
    legalOfficerAddress: string,
    status: LegalOfficerDecisionStatus,
    rejectReason: string | null,
    decisionOn: string | null,
}

export interface ProtectionRequest {
    id: string,
    requesterAddress: string,
    decisions: LegalOfficerDecision[],
    userIdentity: Identity,
    userPostalAddress: PostalAddress,
    createdOn: string
}

export const LIGHT_MODE: ColorTheme = {
    type: 'light',
    shadowColor: rgbaToHex('#3b6cf4', 0.1),
    dashboard: {
        background: rgbaToHex('#3b6cf4', 0.05),
        foreground: '#000000',
    },
    sidebar: {
        background: '#ffffff',
        foreground: '#000000',
        activeItemBackground: rgbaToHex('#3b6cf4', 0.15),
    },
    accounts: {
        iconBackground: '#3b6cf4',
        hintColor: rgbaToHex('#000000', 0.6),
        foreground: '#000000',
        background: rgbaToHex('#3b6cf4', 0.05),
        legalOfficerIcon: {
            category: 'legal-officer',
            id: 'account-shield'
        }
    },
    frame: {
        background: '#ffffff',
        foreground: '#000000',
        link: rgbaToHex('#3b6cf4', 0.15),
    },
    topMenuItems: {
        iconGradient: {
            from: '#3b6cf4',
            to: '#6050dc',
        }
    },
    bottomMenuItems: {
        iconGradient: {
            from: '#7a90cb',
            to: '#3b6cf4',
        }
    }
};
