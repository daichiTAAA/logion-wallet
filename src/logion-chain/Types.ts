import BN from 'bn.js';

export type AssetId = BN;

export type AssetBalance = BN;

export interface AssetMetadata {
    name: string,
    symbol: string,
    decimals: number
}

export interface Asset {
    assetId: AssetId,
    issuer: string,
    metadata: AssetMetadata,
}

export interface AssetWithBalance {
    asset: Asset,
    balance: string,
}
