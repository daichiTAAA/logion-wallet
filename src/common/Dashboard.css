.Dashboard {
    min-height: 100%;
    position: relative;
    display: flex;
}

.Dashboard .Sidebar {
    display: flex;
    flex-direction: column;
    min-height: 100vh;
    height: 100%;
    width: 280px;
}

.Dashboard .Sidebar .MenuArea {
    height: calc(100vh - 160px);
    display: flex;
    flex-direction: column;
    justify-content: space-between;
    padding-top: 30px;
    padding-bottom: 40px;
}

.Dashboard .BasePane {
    padding-left: 0;
    padding-right: 10px;
    overflow-x: hidden;
    width: calc(100% - 280px);
}

.Dashboard .ContentArea {
    height: calc(100vh - 160px);
    overflow-y: auto;
}

.Dashboard .PrimaryArea > [class*="row"] {
    margin-bottom: 40px;
}

.Dashboard .PrimaryArea > [class*="row"]:last-child {
    margin-bottom: 0;
}

.Dashboard .primary-area-col {
    padding-left: 0;
    padding-right: 0;
}

.Dashboard .PrimaryArea {
    font-size: 0.9rem;
    overflow-x: hidden;
    padding-left: 40px;
    padding-right: 40px;
    padding-bottom: 30px;
}

.Dashboard .SecondaryArea {
    font-size: 0.9rem;
    overflow-x: hidden;
    padding-top: 23px;
    padding-left: 20px;
    padding-right: 25px;
    padding-bottom: 30px;
}

.Dashboard .BasePaneHeader {
    display: flex;
}

.Dashboard .BasePaneHeader .TitleArea {
    width: calc(100% - 400px);
}

.Dashboard .BasePaneHeader .AddressSwitcherArea {
    display: flex;
    justify-content: flex-end;
    padding-top: 30px;
    padding-bottom: 30px;
    padding-right: 20px;
    padding-left: 20px;
}

.Dashboard .PrimaryArea h1,
.Dashboard .SecondaryArea h1 {
    font-size: 1.2rem;
    font-weight: bold;
    padding-top: 30px;
    padding-bottom: 60px;
}

.Dashboard .PrimaryArea h2,
.Dashboard .SecondaryArea h2 {
    font-size: 1.1rem;
    font-weight: bold;
}

.Dashboard .PrimaryArea h3,
.Dashboard .SecondaryArea h3 {
    font-size: 1rem;
    padding-bottom: 20px;
}

.Dashboard .PrimaryArea p,
.Dashboard .SecondaryArea p {
    font-size: 0.9rem;
    font-weight: normal;
}

.Dashboard .Titles {
    padding-left: 40px;
    height: 160px;
}

.Dashboard .Titles h1 {
    font-size: 1.5rem;
    font-weight: bold;
    padding-top: 30px;
    white-space: nowrap;
    overflow: hidden;
    text-overflow: ellipsis;
}

.Dashboard .Titles h1 .MenuIcon {
    display: inline-block;
}

.Dashboard .Titles h1 .MenuIcon img {
    vertical-align: top;
    max-height: 36px;
}

.Dashboard .Titles h2 {
    font-size: 1.2rem;
    padding-top: 20px;
    margin-left: 60px;
    white-space: nowrap;
    overflow: hidden;
    text-overflow: ellipsis;
}

.Dashboard .back-button {
    margin-top: 30px;
    font-size: 18px;
    font-weight: bold;
}

.Dashboard .back-button img {
    margin-right: 10px;
}

.Dashboard .TopArea {
    font-size: 0.9rem;
    overflow-x: hidden;
    padding-top: 23px;
    padding-left: 30px;
    padding-right: 40px;
    padding-bottom: 30px;
}

@media (max-width: 1350px) {
    .Dashboard .Sidebar {
        width: 260px;
    }
    .Dashboard .BasePane {
        width: calc(100% - 260px);
    }    
}
