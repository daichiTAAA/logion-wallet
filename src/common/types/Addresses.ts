import moment, { Moment } from 'moment';
import { InjectedAccountWithMeta } from '@polkadot/extension-inject/types';
import { isLegalOfficer } from './LegalOfficer';

export interface Token {
    readonly value: string;
    readonly expirationDateTime: Moment;
}

export interface AccountAddress {
    readonly name: string,
    readonly address: string,
    readonly isLegalOfficer: boolean,
    readonly token?: Token,
}

export default interface Addresses {
    readonly addresses: AccountAddress[],
    readonly currentAddress?: AccountAddress,
}

export type AccountTokens = Record<string, Token>;

export function buildAddresses(
    injectedAccounts: InjectedAccountWithMeta[],
    userAddress: string | undefined,
    tokens: AccountTokens
): Addresses {
    const selectedAddress = currentOrDefaultAddress(injectedAccounts, userAddress, tokens);

    let currentAddress: AccountAddress | undefined;
    const matchingAddress = injectedAccounts.find(injectedAccount => injectedAccount.address === selectedAddress);
    if(selectedAddress !== undefined && matchingAddress !== undefined) {
        currentAddress = {
            name: matchingAddress.meta.name!,
            address: selectedAddress!,
            isLegalOfficer: isLegalOfficer(selectedAddress),
            token: tokenOrUndefinedIfExpired(tokens[selectedAddress]),
        }
    } else {
        currentAddress = undefined;
    }

    return {
        currentAddress,
        addresses: injectedAccounts.map(injectedAccount => ({
            name: injectedAccount.meta.name!,
            address: injectedAccount.address,
            isLegalOfficer: isLegalOfficer(injectedAccount.address),
            token: tokenOrUndefinedIfExpired(tokens[injectedAccount.address]),
        }))
    }
}

function currentOrDefaultAddress(
    injectedAccounts: InjectedAccountWithMeta[],
    currentAddress: string | undefined,
    tokens: AccountTokens
): string | undefined {
    const loggedAddresses = Object.keys(tokens);
    if(currentAddress !== undefined) {
        return currentAddress;
    } else if(loggedAddresses.length > 0) {
        const defaultAccount = injectedAccounts.find(account => loggedAddresses.includes(account.address));
        return defaultAccount?.address;
    } else {
        return undefined;
    }
}

function tokenOrUndefinedIfExpired(token: Token | undefined): Token | undefined {
    if(token === undefined) {
        return undefined;
    }

    const expirationDateTime = token.expirationDateTime;
    if(expirationDateTime.isBefore(moment())) {
        return undefined;
    } else {
        return token;
    }
}
