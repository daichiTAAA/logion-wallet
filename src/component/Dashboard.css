.Dashboard {
    min-height: 100%;
}

.Dashboard .Sidebar {
    display: flex;
    flex-direction: column;
    height: 100vh;
}

.Dashboard .Sidebar .MenuArea {
    height: calc(100vh - 160px);
    display: flex;
    flex-direction: column;
    justify-content: space-between;
    margin-left: -15px;
    margin-right: -15px;
    padding-top: 30px;
    padding-bottom: 40px;
}

.Dashboard .PrimaryArea {
    padding-left: 40px;
    padding-right: 40px;
    font-size: 14px;
    overflow-x: hidden;
}

.Dashboard .SecondaryArea {
    font-size: 14px;
    overflow-x: hidden;
    min-height: 100vh;
}

.Dashboard .SecondaryArea .AddressSwitcherArea {
    padding-top: 30px;
    padding-bottom: 30px;
    padding-right: 20px;
}

.Dashboard .PrimaryArea h1,
.Dashboard .SecondaryArea h1 {
    font-size: 36px;
    font-weight: bold;
    padding-top: 30px;
    padding-bottom: 60px;
}

.Dashboard .PrimaryArea h2,
.Dashboard .SecondaryArea h2 {
    font-size: 18px;
    font-weight: bold;
}

.Dashboard .PrimaryArea h3,
.Dashboard .SecondaryArea h3 {
    font-size: 16px;
    font-weight: bold;
}

.Dashboard .PrimaryArea p,
.Dashboard .SecondaryArea p {
    font-size: 14px;
    font-weight: normal;
}
