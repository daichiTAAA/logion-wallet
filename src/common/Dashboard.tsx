import React from 'react';
import Container from 'react-bootstrap/Container';
import Row from 'react-bootstrap/Row';
import Col from 'react-bootstrap/Col';

import { Children } from './types/Helpers';
import Logo from './Logo';
import AddressSwitcher from './AddressSwitcher';
import { MenuIcon as MenuIconType } from './ColorTheme';
import Menu from './Menu';
import { MenuItemData } from './MenuItem';
import MainMenu from './MainMenu';
import MenuIcon from './MenuIcon';
import Clickable from './Clickable';
import Icon from './Icon';
import { useCommonContext } from './CommonContext';

import './Dashboard.css';

interface TopAreaProps {
    children: Children,
}

function TopArea(props: TopAreaProps) {

    return (
        <Row>
            <Col
                key="top-area"
                className="top-area-col"
            >
                <div className="TopArea">
                    { props.children }
                </div>
            </Col>
        </Row>
    );
}

interface PrimaryAreaProps {
    children: Children,
    width: number,
}

function PrimaryArea(props: PrimaryAreaProps) {

    return (
        <Col
            key="primary-area"
            md={ props.width }
            className="primary-area-col"
        >
            <div className="PrimaryArea">
                { props.children }
            </div>
        </Col>
    );
}

interface SecondaryAreaProps {
    children?: Children,
    width: number,
}

function SecondaryArea(props: SecondaryAreaProps) {

    return (
        <Col
            key="secondary-area"
            md={ props.width }
        >
            <div className="SecondaryArea">
                { props.children }
            </div>
        </Col>
    );
}

export interface ContentPaneProps extends TitlesProps {
    topAreaChildren?: Children,
    primaryAreaChildren: Children,
    secondaryAreaChildren: Children,
    primaryPaneWidth?: number,
}

const FULL_WIDTH = 12;
const SIDEBAR_WIDTH = 2;

interface TitlesProps {
    mainTitle: string,
    subTitle?: string,
    titleIcon: MenuIconType,
    onBack?: () => void
}

function Titles(props: TitlesProps) {

    return (
        <div className="TitlesArea">
            <h1>
                <MenuIcon
                    { ...props.titleIcon }
                />
                { props.mainTitle }
            </h1>
            {
                props.subTitle !== undefined &&
                <h2>{ props.subTitle }</h2>
            }
            {
                props.onBack !== undefined &&
                <div className="back-button">
                    <Clickable onClick={ props.onBack }>
                        <Icon icon={{id:"back", hasVariants: true}} /> Back
                    </Clickable>
                </div>
            }
        </div>
    );
}

export interface BasePaneProps extends TitlesProps {
    children: Children,
    className?: string,
    topAreaChildren?: Children,
}

function BasePane(props: BasePaneProps) {
    const { selectAddress } = useCommonContext();

    let contentAreaClass = "ContentArea";
    if(props.className !== undefined) {
        contentAreaClass = contentAreaClass + " " + props.className;
    }

    return (
        <Col md={ FULL_WIDTH - SIDEBAR_WIDTH } className="BasePane">
            <Row>
                <Col md={ 8 }>
                    <Titles {...props} />
                </Col>
                <Col md={ 4 }>
                    <div className="AddressSwitcherArea">
                        <AddressSwitcher
                            selectAddress={ selectAddress }
                        />
                    </div>
                </Col>
            </Row>
            {
                props.topAreaChildren &&
                <TopArea>
                    { props.topAreaChildren }
                </TopArea>
            }
            <Row className={ contentAreaClass }>
                { props.children }
            </Row>
        </Col>
    );
}

export function ContentPane(props: ContentPaneProps) {

    let primaryPaneWidth: number = 6;
    if(props.primaryPaneWidth !== undefined) {
        primaryPaneWidth = props.primaryPaneWidth;
    }

    return (
        <BasePane
            mainTitle={ props.mainTitle }
            subTitle={ props.subTitle }
            titleIcon={ props.titleIcon }
            onBack={ props.onBack }
            topAreaChildren={ props.topAreaChildren }
        >
            <PrimaryArea
                width={ primaryPaneWidth }
            >
                { props.primaryAreaChildren }
            </PrimaryArea>
            <SecondaryArea
                width={ FULL_WIDTH - primaryPaneWidth }
            >
                { props.secondaryAreaChildren }
            </SecondaryArea>
        </BasePane>
    );
}

export interface FullWidthPaneProps extends BasePaneProps {
    children: Children,
    className?: string,
}

export function FullWidthPane(props: FullWidthPaneProps) {

    return (
        <BasePane
            className={ props.className }
            mainTitle={ props.mainTitle }
            subTitle={ props.subTitle }
            titleIcon={ props.titleIcon }
            onBack={ props.onBack }
            topAreaChildren={ props.topAreaChildren }
        >
            <PrimaryArea width={ FULL_WIDTH }>
                { props.children }
            </PrimaryArea>
        </BasePane>
    );
}

export interface Props {
    children: Children,
    menuTop: MenuItemData[],
    menuMiddle: MenuItemData[],
    menuBottom: MenuItemData[],
}

export default function Dashboard(props: Props) {
    const { colorTheme } = useCommonContext();

    const inlineCss = `
    .Dashboard .PrimaryArea .table {
        color: ${colorTheme.dashboard.foreground};
    }

    .Dashboard .SecondaryArea .table {
        color: ${colorTheme.dashboard.foreground};
    }

    .Dashboard a {
        color: ${colorTheme.dashboard.foreground};
    }

    .Dashboard .Sidebar .MenuItem.active {
        background-color: ${colorTheme.sidebar.activeItemBackground};
    }
    `;

    return (
        <Container
            className="Dashboard"
            fluid
            style={{
                backgroundColor: colorTheme.dashboard.background,
                color: colorTheme.dashboard.foreground,
            }}
        >
            <style>
            { inlineCss }
            </style>
            <Row>
                <Col md={ SIDEBAR_WIDTH }>
                    <div className="Sidebar"
                        style={{
                            backgroundColor: colorTheme.sidebar.background,
                            color: colorTheme.sidebar.foreground,
                            boxShadow: `0 0 25px ${colorTheme.shadowColor}`,
                        }}
                    >
                        <Logo
                            shadowColor={ colorTheme.shadowColor }
                        />
                        <div
                            className="MenuArea"
                        >
                            <Menu
                                items={ props.menuTop }
                            />
                            <MainMenu
                                items={ props.menuMiddle }
                            />
                            <Menu
                                items={ props.menuBottom }
                            />
                        </div>
                    </div>
                </Col>
                { props.children }
            </Row>
        </Container>
    );
}
