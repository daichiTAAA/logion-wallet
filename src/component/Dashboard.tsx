import React from 'react';
import Container from 'react-bootstrap/Container';
import Row from 'react-bootstrap/Row';
import Col from 'react-bootstrap/Col';

import { Children, BackgroundAndForegroundColors, GradientData } from './types/Helpers';
import Addresses from './types/Addresses';
import Logo from './Logo';
import AddressSwitcher from './AddressSwitcher';
import { AccountAddressColors } from './AccountAddress';
import Menu from './Menu';
import { MenuItemData } from './MenuItem';
import Shield from './Shield';

import './Dashboard.css';

export interface PrimaryAreaColors extends BackgroundAndForegroundColors {
    link: string,
}

export interface MenuAreaColors extends BackgroundAndForegroundColors {
    logoShadow: string,
}

export interface MenuItemColors {
    iconGradient: GradientData,
}

export interface ColorTheme {
    dashboard: BackgroundAndForegroundColors,
    menuArea: MenuAreaColors,
    primaryArea: PrimaryAreaColors,
    secondaryArea: BackgroundAndForegroundColors,
    accounts: AccountAddressColors,
    frame: BackgroundAndForegroundColors,
    topMenu: MenuItemColors,
    bottomMenu: MenuItemColors,
}

export interface PrimaryAreaProps {
    children: Children,
    colors: PrimaryAreaColors,
}

export function PrimaryArea(props: PrimaryAreaProps) {

    return (
        <Col
            key="primary-area"
            md={7}
            style={{
                backgroundColor: props.colors.background,
                color: props.colors.foreground,
            }}
        >
            <div className="PrimaryArea">
                { props.children }
            </div>
        </Col>
    );
}

export interface SecondaryAreaProps {
    children?: Children,
    colors: BackgroundAndForegroundColors,
    addresses: Addresses,
    accountColors: AccountAddressColors,
    selectAddress: (userAddress: string) => void,
}

export function SecondaryArea(props: SecondaryAreaProps) {

    return (
        <Col
            key="secondary-area"
            md={3}
            style={{
                backgroundColor: props.colors.background,
                color: props.colors.foreground,
            }}
        >
            <div className="SecondaryArea">
                <div className="AddressSwitcherArea">
                    <AddressSwitcher
                        addresses={ props.addresses }
                        colors={ props.accountColors }
                        selectAddress={ props.selectAddress }
                    />
                </div>
                { props.children }
            </div>
        </Col>
    );
}

export interface ContentPaneProps {
    addresses: Addresses,
    selectAddress: (userAddress: string) => void,
    primaryAreaChildren: Children,
    secondaryAreaChildren?: Children,
    colors: ColorTheme,
}

export function ContentPane(props: ContentPaneProps) {

    return (
        <>
            <PrimaryArea
                colors={ props.colors.primaryArea }
            >
                { props.primaryAreaChildren }
            </PrimaryArea>
            <SecondaryArea
                colors={ props.colors.secondaryArea }
                addresses={ props.addresses }
                accountColors={ props.colors.accounts }
                selectAddress={ props.selectAddress }
            >
                { props.secondaryAreaChildren }
            </SecondaryArea>
        </>
    );
}

export interface Props {
    children: Children,
    colors: ColorTheme,
    addresses: Addresses,
    selectAddress: (userAddress: string) => void,
    menuTop: MenuItemData[],
    menuBottom: MenuItemData[],
    shieldItem: MenuItemData,
}

export default function Dashboard(props: Props) {

    const inlineCss = `
    .Dashboard .PrimaryArea a,
    .Dashboard .PrimaryArea .btn-link {
        color: ${props.colors.primaryArea.link}
    }

    .Dashboard .PrimaryArea .table {
        color: ${props.colors.primaryArea.foreground}
    }

    .Dashboard .SecondaryArea .table {
        color: ${props.colors.secondaryArea.foreground}
    }
    `;

    return (
        <Container
            className="Dashboard"
            fluid
            style={ props.colors.dashboard }
        >
            <style>
            { inlineCss }
            </style>
            <Row>
                <Col md={2}>
                    <div className="Sidebar">
                        <Logo
                            foreground={ props.colors.menuArea.foreground }
                            background={ props.colors.menuArea.background }
                            logoShadow={ props.colors.menuArea.logoShadow }
                        />
                        <div
                            className="MenuArea"
                            style={{
                                backgroundColor: props.colors.menuArea.background,
                                color: props.colors.menuArea.foreground,
                            }}
                        >
                            <Menu
                                items={ props.menuTop }
                            />
                            <Shield item={ props.shieldItem } />
                            <Menu
                                items={ props.menuBottom }
                            />
                        </div>
                    </div>
                </Col>
                { props.children }
            </Row>
        </Container>
    );
}
