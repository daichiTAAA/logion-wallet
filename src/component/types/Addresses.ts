import { InjectedAccountWithMeta } from '@polkadot/extension-inject/types';

export interface AccountAddress {
    name: string,
    address: string,
}

export default interface Addresses {
    addresses: AccountAddress[],
    currentAddress: AccountAddress,
}

export function buildAddresses(injectedAccounts: InjectedAccountWithMeta[], userAddress: string): Addresses {
    return {
        currentAddress: {
            name: injectedAccounts.filter(injectedAccount => injectedAccount.address === userAddress)[0].meta.name!,
            address: userAddress,
        },
        addresses: injectedAccounts.map(injectedAccount => { return {
            name: injectedAccount.meta.name!,
            address: injectedAccount.address,
        };})
    }
}
