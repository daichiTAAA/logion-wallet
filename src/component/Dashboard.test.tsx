import { shallowRender } from '../tests';

import Dashboard, { ContentPane, FullWidthPane } from './Dashboard';
import { LIGHT_MODE } from '../legal-officer/Types';
import Addresses from './types/Addresses';

const ADDRESSES: Addresses = {
    currentAddress: {
        name: "Name 1",
        address: "address1",
        isLegalOfficer: false,
    },
    addresses: [
        {
            name: "Name 1",
            address: "address1",
            isLegalOfficer: false,
        },
        {
            name: "Name 2",
            address: "address2",
            isLegalOfficer: false,
        }
    ]
}

test("renders Dashboard", () => {
    const result = shallowRender(
        <Dashboard
            colors={ LIGHT_MODE }
            addresses={ ADDRESSES }
            selectAddress={ jest.fn() }
            menuTop={ [] }
            menuBottom={ [] }
            shieldItem={{
                text: "Protection",
                to: "/",
                exact: true,
            }}
        >
        </Dashboard>
    );
    expect(result).toMatchSnapshot();
});

test("renders ContentPane", () => {
    const result = shallowRender(
        <ContentPane
            primaryAreaChildren={ null }
            secondaryAreaChildren={ null }
        />
    );
    expect(result).toMatchSnapshot();
});

test("renders FullWidthPane", () => {
    const result = shallowRender(
        <FullWidthPane>
        </FullWidthPane>
    );
    expect(result).toMatchSnapshot();
});
