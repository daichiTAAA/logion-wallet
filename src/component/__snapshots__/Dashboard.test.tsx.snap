// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders ContentPane 1`] = `
<BasePane>
  <PrimaryArea
    width={6}
  />
  <SecondaryArea
    width={6}
  />
</BasePane>
`;

exports[`renders Dashboard 1`] = `
<Container
  className="Dashboard"
  fluid={true}
  style={
    Object {
      "backgroundColor": "#eff3fe",
      "color": "#000000",
    }
  }
>
  <style>
    
    .Dashboard .PrimaryArea .table {
        color: #000000;
    }

    .Dashboard .SecondaryArea .table {
        color: #000000;
    }

    .modal-dialog {
        background-color: white;
    }

    .modal-content {
        color: #000000;
        background-color: #eff3fe;
    }

    .Dashboard .form-control,
    .Dashboard .form-control[readonly] {
        background-color: #eff3fe;
        color: #000000;
    }

    .Dashboard a {
        color: #000000;
    }

    .Dashboard .Sidebar .MenuItem.active {
        background-color: #d8e2fd;
    }
    
  </style>
  <Row
    noGutters={true}
  >
    <Col
      md={2}
    >
      <div
        className="Sidebar"
        style={
          Object {
            "backgroundColor": "#ffffff",
            "boxShadow": "0 0 25px #3b6cf41a",
            "color": "#000000",
          }
        }
      >
        <Logo
          colorThemeType="light"
          shadowColor="#3b6cf41a"
        />
        <div
          className="MenuArea"
        >
          <Menu
            colorThemeType="light"
            items={Array []}
          />
          <MainMenu
            colorThemeType="light"
          />
          <Menu
            colorThemeType="light"
            items={Array []}
          />
        </div>
      </div>
    </Col>
  </Row>
</Container>
`;

exports[`renders FullWidthPane 1`] = `
<BasePane>
  <PrimaryArea
    width={12}
  />
</BasePane>
`;
